### PR TITLE
Add an option to disable keyring file

### LINKED ISSUE #835: Don't persist gossip encryption keys
A common security recommendation is to store credentials and secret keys on a ramdisk so that they go away if the host is rebooted and are never persisted.

Would it be possible to (optionally) store the keyring state only in memory? With this enabled, the agent would always use the provided key on start, trusting that it will be the current key, then get the rest of the keyring from other nodes.

An external system would be responsible for ensuring that each node has the correct key to use on start.

One workaround for not having this feature is putting the entire serf directory on a ramdisk. Would that cause any issues? Is there state in the serf directory that needs to be persisted across reboots?

### PR DESCRIPTION
**Pull request by kyhavlov**
Also disables keyring file in dev mode.

Closes #835, #2630, and #2784.

### CODE REVIEW

File: vendor/vendor.json [hunk 1]
```diff
@@ -675,11 +675,11 @@
 			"revisionTime": "2017-05-25T23:15:04Z"
 		},
 		{
-			"checksumSHA1": "ZkJRgexeNzNZzpw6YnedwoJl7pE=",
+			"checksumSHA1": "sdjkMflUxYtL/l7n7wYPS0Xj5c0=",
```
Comment (preetapan) [inline vendor/vendor.json]: Curious, why did vendor.json and the above two files have to change for this, they seem unrelated to the change being made?

File: agent/agent.go [hunk 2]
```diff
@@ -887,8 +887,10 @@ func (a *Agent) makeServer() (*consul.Server, error) {
 	if err != nil {
 		return nil, err
 	}
-	if err := a.setupKeyrings(config); err != nil {
-		return nil, fmt.Errorf("Failed to configure keyring: %v", err)
+	if !a.config.DisableKeyringFile {
```
Comment (preetapan) [inline agent/agent.go]: Minor readability thing - Any reason why this config param is not enableKeyRingFile, set to false by default? I find negative boolean feature flags always take a couple of seconds more to follow when you are reading a config file that has them, because its like if its set to true the file won't be written

Comment (slackpad) [inline agent/agent.go]: This isn't a bad idea since we are creating the config and not worrying about BC. If we switch to a `bool*` we could make this `EnableKeyringFile` and default it to `true`.

File: agent/config_test.go [hunk 3]
```diff
@@ -318,6 +318,10 @@ func TestDecodeConfig(t *testing.T) {
 			c:  &Config{EnableSyslog: true},
 		},
 		{
+			in: `{"disable_keyring_file":true}`,
```
Comment (slackpad) [inline agent/config_test.go]: Need to update the website with the new option.

File: agent/agent.go (hunk 2 above)
Comment (slackpad) [inline agent/agent.go]: I'm ok leaving it disable too, since it defaults to enabled and we have a lot of other things like that in the config. I'll leave it up to you @kyhavlov - either way works.
